### ISSUE TITLE
lopp for novetly detection

### ISSUE THREAD

**Issue by RockLeeStudio** [vc1492a/PyNomaly#31]
lof can be used to predict unseen data using predict method.
I can't find predict method in lopp. Does lopp support this scenario?

**Comment by vc1492a** [comment]
Hi @RockLeeStudio, thanks for opening this issue. You are correct that the older implementations of Local Outlier Factor (LOF) available in scikit-learn as well as some of the versions of pyod were available with a fit_predict() or predict() method. However, [pyod is deprecating fit_predict() as of v0.6.9 and removing it altogether in v7.2.0](https://pyod.readthedocs.io/en/latest/pyod.models.html#pyod.models.lof.LOF.fit_predict) and I no longer see such a method in the [latest scikit-learn documentation](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.LocalOutlierFactor.html). I can't speak to the reason's why pyod and scikit-learn is/have deprecating/ed their fit_predict() method, but the reasons why those methods are not included in PyNomaly's implementation is a matter of design choice. 

As an unsupervised approach, Local Outlier Probabilities is not trained and used to predict on unseen data, such as a test set. It was intended to be fit over the entirety of the data available. While the streaming approach that's available in PyNomaly overcomes this limitation to some degree (still needs periodic retraining of the original approach), fundamentally the methodology is intended to fit over all the data in which you'd like to assign a score. 

Now to how the implementation used to work (at a summary level) in scikit-learn. In their implementation, some thresholding algorithm (I was at one point familiar with how this worked but haven't examined the code for a while) is used to select the decision boundary (the value in which what would be considered an outlier, e.g. 0.96). This value is then used as a cut-off point to assign a 1 or a 0 as to whether the point is considered an anomaly or outlier. 

The choice of value to select is dependent on the data used as input and the parameters selected, ans is highly variable. There are no plans to introduce a predict() method in PyNomaly because of this, so I'll close this issue. You could however introduce your own approach to determining when a value should be considered as an outlier and use the values, available after fitting as the attribute `. local_outlier_probabilities`, as an input to your scoring approach. 

Hope this helps!

**Comment by RockLeeStudio** [comment]
Thanks for your insightful reply.It's very helpful.
